PermissionError on Windows systems during shutdown

### First Check

- [x] I added a very descriptive title here.
- [x] This is not a security issue (those should be reported via the [security advisory](https://github.com/zauberzeug/nicegui/security/advisories/new) instead).
- [x] This is not a Q&A. I am sure something is wrong with NiceGUI or its documentation.
- [x] I used the GitHub search to find a similar [issue](https://github.com/zauberzeug/nicegui/issues) and came up empty.

### Example Code

```python
# any type of test using the storage should do, but since this is non-deterministic, it's hard to verify
                                                                                                              
from nicegui import app, ui                                                                                                                                                 
from nicegui.testing import User    
import pytest                                                                                                                                          
                                                                                                                                                                              
@ui.page("/")
def index() -> None:
    ui.label("hello")

if __name__ in {"__main__", "__mp_main__"}:
    ui.run()                                                                                                                                                        
   
@pytest.mark.nicegui_main_file(__file__)
async def test_storage_write(user: User) -> None:                                                                                                                           
    await user.open("/")
    app.storage.general["key"] = "value"  
```

### Description

When running some tests on Windows machines (with the Github runners), an error came up non-deterministically:

```
_______ ERROR at teardown of test_ok_validation_no_filename_write_mode ________

    def finalizer() -> None:
        """Yield again, to finalize."""
    
        async def async_finalizer() -> None:
            try:
                await gen_obj.__anext__()
            except StopAsyncIteration:
                pass
            else:
                msg = "Async generator fixture didn't stop."
                msg += "Yield only once."
                raise ValueError(msg)
    
>       runner.run(async_finalizer(), context=context)

.tox\3.14\Lib\site-packages\pytest_asyncio\plugin.py:330: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
C:\hostedtoolcache\windows\Python\3.14.3\x64\Lib\asyncio\runners.py:127: in run
    return self._loop.run_until_complete(task)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
C:\hostedtoolcache\windows\Python\3.14.3\x64\Lib\asyncio\base_events.py:719: in run_until_complete
    return future.result()
           ^^^^^^^^^^^^^^^
.tox\3.14\Lib\site-packages\pytest_asyncio\plugin.py:322: in async_finalizer
    await gen_obj.__anext__()
.tox\3.14\Lib\site-packages\nicegui\testing\user_plugin.py:25: in user
    async with user_simulation(main_file=get_path_to_main_file(request)) as user:
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
C:\hostedtoolcache\windows\Python\3.14.3\x64\Lib\contextlib.py:221: in __aexit__
    await anext(self.gen)
.tox\3.14\Lib\site-packages\nicegui\testing\user_simulation.py:31: in user_simulation
    with nicegui_reset_globals():
         ^^^^^^^^^^^^^^^^^^^^^^^
C:\hostedtoolcache\windows\Python\3.14.3\x64\Lib\contextlib.py:148: in __exit__
    next(self.gen)
.tox\3.14\Lib\site-packages\nicegui\testing\general.py:68: in nicegui_reset_globals
    app.reset()
.tox\3.14\Lib\site-packages\nicegui\app\app.py:360: in reset
    self.storage.clear()
.tox\3.14\Lib\site-packages\nicegui\storage.py:196: in clear
    self._general.clear()
.tox\3.14\Lib\site-packages\nicegui\persistence\file_persistent_dict.py:58: in clear
    self.filepath.unlink(missing_ok=True)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = WindowsPath('D:/a/NiceGooey/NiceGooey/.nicegui/storage-general.json')
missing_ok = True

    def unlink(self, missing_ok=False):
        """
        Remove this file or link.
        If the path is a directory, use rmdir() instead.
        """
        try:
>           os.unlink(self)
E           PermissionError: [WinError 32] The process cannot access the file because it is being used by another process: 'D:\\a\\NiceGooey\\NiceGooey\\.nicegui\\storage-general.json'

C:\hostedtoolcache\windows\Python\3.14.3\x64\Lib\pathlib\__init__.py:1042: PermissionError
```


I haven't found a workaround or possible solution yet.

### NiceGUI Version

3.10.0

### Python Version

3.14.3

### Browser

Firefox

### Operating System

Windows

### Additional Context

_No response_

--- a/nicegui/persistence/file_persistent_dict.py
+++ b/nicegui/persistence/file_persistent_dict.py
@@ -45,14 +45,15 @@ class FilePersistentDict(PersistentDict):
 
         @background_tasks.await_on_shutdown
         async def async_backup() -> None:
+            if not self:
+                self.filepath.unlink(missing_ok=True)
+                return
             async with aiofiles.open(self.filepath, 'w', encoding=self.encoding) as f:
                 await f.write(json.dumps(self, indent=self.indent))
 
         if core.is_loop_running():
             background_tasks.create_lazy(async_backup(), name=self.filepath.stem)
+        elif not self:
+            self.filepath.unlink(missing_ok=True)
         else:
             self.filepath.write_text(json.dumps(self, indent=self.indent), encoding=self.encoding)
-
-    def clear(self) -> None:
-        super().clear()
-        self.filepath.unlink(missing_ok=True)

